src/codeu/chat/client/maingui/ConversationPanel.java

```diff
@@ -0,0 +1,137 @@
+package codeu.chat.client.maingui;
+
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import javax.swing.*;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+
+import codeu.chat.client.ClientContext;
+import codeu.chat.common.ConversationSummary;
+
+// NOTE: JPanel is serializable, but there is no need to serialize ConversationPanel
+// without the @SuppressWarnings, the compiler will complain of no override for serialVersionUID
+@SuppressWarnings("serial")
+public final class ConversationPanel extends JPanel {
+
+  private final ClientContext clientContext;
+  private final MessagePanel messagePanel;
+
+  public ConversationPanel(ClientContext clientContext, MessagePanel messagePanel) {
+    super();
+    this.clientContext = clientContext;
+    this.messagePanel = messagePanel;
+    initialize();
+  }
+
+  private void initialize() {
+
+    // This panel contains from top to bottom: a title bar,
+    // a list of conversations, and a button bar.
+    this.setLayout(new BorderLayout());
+    this.setPreferredSize(new Dimension(200,0));
+    this.setBorder(BorderFactory.createMatteBorder(0,0,0,1,Color.BLACK));
+
+    // Title
+    final JPanel titlePanel = new JPanel(new FlowLayout(FlowLayout.CENTER));
+
+    final JLabel titleLabel = new JLabel("Conversations", JLabel.CENTER);
+    titleLabel.setAlignmentX(Component.LEFT_ALIGNMENT);
+    titlePanel.add(titleLabel);
+
+    // Conversation list
+    final JPanel listShowPanel = new JPanel(new BorderLayout());
+
+    final DefaultListModel<String> listModel = new DefaultListModel<>();
+    final JList<String> objectList = new JList<>(listModel);
+    objectList.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+    objectList.setSelectedIndex(-1);
+
+    final JScrollPane listScrollPane = new JScrollPane(objectList);
+    listShowPanel.add(listScrollPane, BorderLayout.CENTER);
+
+    // Button bar
+    final JPanel buttonPanel = new JPanel();
+
+    final JButton updateButton = new JButton("Update");
+    final JButton addButton = new JButton("Add");
+
+    updateButton.setAlignmentX(Component.LEFT_ALIGNMENT);
+    buttonPanel.add(updateButton);
+    buttonPanel.add(addButton);
+
+    this.add(titlePanel, BorderLayout.NORTH);
+    this.add(listShowPanel, BorderLayout.CENTER);
+    this.add(buttonPanel, BorderLayout.SOUTH);
+
```

Comment:
This is the first time I see someone order the UI elements in such an easy-to-understand and organized way. Nice.